react/src/components/atoms/forms/InputCurrency/index.js

```diff
@@ -47,148 +47,130 @@ const Currency = (props) => {
-                  }, () => {
-                    if (typeof props.onChange === 'function') {
+                if (greaterThanMin(newValue) && lessThanMax(newValue)) {
+                  const updateError = displayErrorMessage(!is.empty(stringValue) ? newValue : '');
+                  context.updateState({ value: toCurrency(newValue, countDecimals(props.step)), ...updateError }, () => {
+                    if (is.fn(props.onChange)) {
                       props.onChange(newValue, props.id, type, key);
                     }
                   });
                 }
               } else if (key === 'ArrowUp') {
                 newValue = Number(numbro(numberValue).add(props.step).format({ mantissa: countDecimals(props.step) }));
-                if ((!hasProperty(props, 'min') || newValue >= props.min) && (!hasProperty(props, 'max') || (newValue <= props.max))) {
-                  const { showError, errorMsg } = validNumber(newValue, props.min, props.max);
-                  context.updateState({
-                    showError,
-                    errorMsg,
-                    value: toCurrency(newValue, countDecimals(props.step))
-                  }, () => {
-                    if (typeof props.onChange === 'function') {
+                if (greaterThanMin(newValue) && lessThanMax(newValue)) {
+                  const updateError = displayErrorMessage(!is.empty(stringValue) ? newValue : '');
+                  context.updateState({ value: toCurrency(newValue, countDecimals(props.step)), ...updateError }, () => {
+                    if (is.fn(props.onChange)) {
                       props.onChange(newValue, props.id, type, key);
                     }
                   });
                 }
               }
             }
           };
+
           const handleBlur = () => {
             const inputEl = ref.current;
-            if (is.empty(inputEl.value)) {
+            const stringValue = inputEl.value;
+            const isNotNumber = !stringValue || isNaN(Number(numbro.unformat(stringValue)));
+            if (isNotNumber) {
               inputEl.setAttribute('placeholder', props.placeholder);
             }
-            const stringValue = inputEl.value;
-            const numberValue = Number(numbro.unformat(stringValue));
+            let newValue = isNotNumber ? '' : Number(numbro.unformat(stringValue));
             if (props.required && is.empty(stringValue)) {
               errorMsg = 'Please enter a value.';
               context.updateState({ showError: true, errorMsg });
             } else if (!is.empty(stringValue)) {
-              let newValue = numberValue;
-              if (hasProperty(props, 'max') && newValue > props.max) {
+              if (!hasNumberProperty(props, 'max') || newValue > props.max) {
                 newValue = props.max;
               }
-              if (hasProperty(props, 'min') && newValue < props.min) {
+              if (!hasNumberProperty(props, 'min') || newValue < props.min) {
```

Comment:
```suggestion
              if (!hasNumberProperty(props, 'min') || (!isNotNumber && newValue < props.min)) {
```

Reply:
this check already happend on https://github.com/massgov/mayflower/pull/518/files#diff-a4f99a33aa0b972fd31cf3cd707ceaf1R147

Reply:
isNotNumber returns true if stringValue is null, undefined or 'NaN'. newValue is either '' or a number value